File: csharp/GetCryptoExchanges.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using Intrinio.SDK.Api;
using Intrinio.SDK.Client;
using Intrinio.SDK.Model;

namespace Example
{
  public class GetCryptoExchanges
  {
    public static void Main()
    {
      Configuration.Default.AddApiKey("api_key", "YOUR_API_KEY");

      var cryptoApi = new CryptoApi();
      var pair = "btcusd";  // string | Returns Crypto Currencies traded on the given Crypto Exchange. (optional) 

      try
      {
        ApiResponseCryptoExchanges result = cryptoApi.GetCryptoExchanges(pair);
        List<CryptoExchange> exchanges = result.Exchanges;

        Console.WriteLine(exchanges.Count + " exchanges found!");
        Console.WriteLine();

        exchanges.ForEach(delegate (CryptoExchange exchange)
        {
          Console.WriteLine("Name:                 " + exchange.Name);
          Console.WriteLine("Code:                 " + exchange.Code);
          Console.WriteLine("Book depth available: " + exchange.BookDepthAvailable);
          Console.WriteLine("History available:    " + exchange.HistoryAvailable);
          Console.WriteLine("Trades available:     " + exchange.TradesAvailable);

          List<String> pairs = exchange.Pairs;
          Console.WriteLine("Pairs:");
          pairs.ForEach(delegate (String crypto_pair)
          {
            Console.WriteLine("  - " + crypto_pair);
          });

          Console.WriteLine();
        });
      }
      catch (Exception e)
      {
        Debug.Print("Exception when calling CryptoApi.GetCryptoExchanges: " + e.Message);
      }
    }
  }
}

File: python/get_crypto_exchanges.py
from __future__ import print_function
import time
import intrinio_sdk
from intrinio_sdk.rest import ApiException
from pprint import pprint
 
intrinio_sdk.ApiClient().configuration.api_key['api_key'] = 'YOUR_API_KEY'
 
crypto_api = intrinio_sdk.CryptoApi()

pair = 'btcusd' # str | Returns Crypto Currencies traded on the given Crypto Exchange. (optional)
 
try:
    api_response = crypto_api.get_crypto_exchanges(pair=pair)
    exchanges = api_response.exchanges

    if pair:
        print("%s exchanges found for %s!" % (len(exchanges), pair))
    else:
        print("%s exchanges found!" % (len(exchanges)))

    print("------------------------------------------------------------")

    for exchange in exchanges:
        print("Name:                 %s" % (exchange.name))
        print("Code:                 %s" % (exchange.code))
        print("Book depth available: %s" % (exchange.book_depth_available))
        print("History available:    %s" % (exchange.history_available))
        print("Snapshot available:   %s" % (exchange.snapshot_available))
        print("Trades available:     %s" % (exchange.trades_available))
        print("")

except ApiException as e:
    print("Exception when calling CryptoApi->get_crypto_exchanges: %s\n" % e)
